### PR TITLE
chore(firebase_storage): fix false positive tests output

### PR DESCRIPTION
**Pull request by Salakar**
## Description

* Tests were not setup correctly, so even when a test failed it would exit as successful.
* Switched to use `e2e`.
* Temporarily disable an http request inside the tests (recently started failing, manual repro works fine, so tests related issue).

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_storage/example/test_driver/firebase_storage.dart [hunk 1]
```diff
@@ -16,6 +17,7 @@ void main() {
     FirebaseStorage firebaseStorage;
 
     setUp(() async {
+      await Firebase.initializeApp();
       firebaseStorage = FirebaseStorage();
```
Comment (kroikie) [inline packages/firebase_storage/example/test_driver/firebase_storage.dart:21]: If storage is migrated to the new core, would this still be the way to get a storage instance? I would expect it to be something like `firebaseStorage = FirebaseStorage.instance`?

Comment (Ehesp) [reply packages/firebase_storage/example/test_driver/firebase_storage.dart:21]: New core is `await Firebase.initializeApp();`, the instance getter for each plugin is being updated when we rework the plugin itself (to handle `.instance` and `instanceFor()`).

Comment (kroikie) [reply packages/firebase_storage/example/test_driver/firebase_storage.dart:21]: OK cool, so the move to the new instance getter will be in a follow up PR?

Comment (Ehesp) [reply packages/firebase_storage/example/test_driver/firebase_storage.dart:21]: Yep, once we move to Storage it'll be handled there.

**Pull request merged**